x/trigger/types/genesis_test.go

```diff
@@ -0,0 +1,276 @@
+				gs.QueuedTriggers[0].Trigger.Owner = "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma"
+				gs.Triggers[0].Id = 5
+			},
+			err: "trigger id is invalid and cannot exceed 2",
+		},
+		{
+			name: "invalid - queued trigger action must pass internal validate basic",
+			state: &GenesisState{
+				TriggerId:      2,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{{TriggerId: 1, Amount: 1}, {TriggerId: 2, Amount: 1}},
+				Triggers:       []Trigger{trigger},
+				QueuedTriggers: []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger2}},
+			},
+			modify: func(gs *GenesisState) {
+				gs.QueuedTriggers[0].Trigger.Actions = badRequest.Actions
+			},
+			err: "empty address string is not allowed",
+		},
+		{
+			name: "invalid - A queued trigger's id cannot exceed state trigger id",
+			state: &GenesisState{
+				TriggerId:      2,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{{TriggerId: 1, Amount: 1}, {TriggerId: 2, Amount: 1}},
+				Triggers:       []Trigger{trigger},
+				QueuedTriggers: []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger2}},
+			},
+			modify: func(gs *GenesisState) {
+				gs.Triggers[0].Owner = "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma"
+				gs.QueuedTriggers[0].Trigger.Owner = "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma"
+				gs.QueuedTriggers[0].Trigger.Id = 5
+			},
+			err: "trigger id is invalid and cannot exceed 2",
+		},
+		{
+			name: "invalid - A trigger's event must pass validation",
+			state: &GenesisState{
+				TriggerId:      1,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{{TriggerId: 1, Amount: 1}, {TriggerId: 2, Amount: 1}},
+				Triggers:       []Trigger{trigger},
+				QueuedTriggers: []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger2}},
+			},
+			modify: func(gs *GenesisState) {
+				gs.Triggers[0].Owner = "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma"
+				gs.Triggers[0].Event = badRequest.Event
+			},
+			err: "empty event name",
+		},
+		{
+			name: "invalid - A queued trigger's event must pass validation",
+			state: &GenesisState{
+				TriggerId:      2,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{{TriggerId: 1, Amount: 1}, {TriggerId: 2, Amount: 1}},
+				Triggers:       []Trigger{trigger},
+				QueuedTriggers: []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger2}},
+			},
+			modify: func(gs *GenesisState) {
+				gs.Triggers[0].Owner = "cosmos1w6t0l7z0yerj49ehnqwqaayxqpe3u7e23edgma"
+				gs.QueuedTriggers[0].Trigger.Event = badRequest.Event
+			},
+			err: "empty event name",
+		},
+		{
+			name: "invalid - Gas limits must match either a trigger or queued trigger",
+			state: &GenesisState{
+				TriggerId:      3,
+				QueueStart:     1,
+				GasLimits:      []GasLimit{{TriggerId: 3, Amount: 1}, {TriggerId: 2, Amount: 1}},
+				Triggers:       []Trigger{trigger},
+				QueuedTriggers: []QueuedTrigger{{BlockHeight: 1, Time: time.Time{}, Trigger: trigger2}},
+			},
+			modify: nil,
```

Comment:
This isn't something that needs fixing, it's just for future reference.

The nice thing about using named fields when defining these is that you can omit things that you don't need, don't care about, or don't apply to a test case.

In this case, you can just leave this `modify: nil,` line out.

One place I do that a lot is when testing functions that can return both a thing and an error, and I expect only one or the other to be returned. In my test case definition, I'll only include the one field, and omit the one that I'm not expecting to get.